Modules Design update (#769)
Azure/ResourceModules: docs/wiki/ModulesDesign.md

Each module should come with a `.bicep` folder with a least the `nested_cuaId.bicep` file in it

## Naming

Use the following naming standard for module files and folders:

- Module folders are in camelCase and their name reflects the main resource type of the Bicep module they are hosting (e.g. `storageAccounts`, `virtualMachines`).

> - Cross-referenced and providers resource modules are placed in the `.bicep` subfolder and named `nested_<crossReferencedResourceType>.bicep`

Reviewer: ```suggestion
- Cross-referenced and provider resource modules are placed in the `.bicep` subfolder and named `nested_<crossReferencedResourceType>.bicep`
```

Reviewer 2: Extension resources... there is nothing in the official doc called provider resource modules.

Author: Replaced providers occurrences in this .md file. Probably we're using this naming in other files and they should be replaced as well.

## Variables

- Variable names are in camelCase, e.g. `builtInRoleNames`.

> - For modules that manage roleAssignments, typically cross-referenced `nested_rbac.bicep` in the `.bicep` folder, update the list of roles to only be the applicable roles. You can find a helper script `Get-FormattedRBACRoles.ps1` in the `utilities\tools\helper` folder of the repository.

Reviewer: ```suggestion
- Modules that manage roleAssignments typically have a cross-referenced `nested_rbac.bicep` file in their `.bicep` folder.  Those files require you to update their list of roles to only the applicable roles. You can find a helper script `Get-FormattedRBACRoles.ps1` in the `utilities/tools/helper` folder of the repository.
```

Reviewer 2: I suggest moving this to the RBAC chapter on this article. Its not general consideration to variables, but this is more specific to RBAC.

Author: Moved and updated

        properties: {...}
      }
      ```
## Modules

  - Module symbolic names are in camel_Snake_Case, following the schema `<mainResourceType>_<referencedResourceType>` e.g. `storageAccount_fileServices`, `virtualMachine_nic`, `resourceGroup_rbac`.

>   - Modules enable you to reuse code from a Bicep file in other Bicep files. As such they're normally leveraged for deploying child resources (e.g. file services in a storage account), cross referenced resources (e.g. network interface in a virtual machine) or providers (e.g. role assignment in a resource group).

Reviewer: ```suggestion
  - Modules enable you to reuse code from one Bicep file in another Bicep file. As such they're usually leveraged to deploy child resources (e.g. file services in a storage account), cross referenced resources (e.g. network interface in a virtual machine) or providers (e.g. role assignment in a resource group).
```

Reviewer 2: Again there is nothing called providers. Call this extension resources

Author: Done.

### Deployment names

> When using modules from parent resources you will need to specify a name that, when deployed, will be used to assign the deployment name.

Reviewer: ```suggestion
When using modules from parent resources you will need to specify a name (i.e. deployment name) that, when deployed, will be used to assign the deployment name.
```

There are some constraints that needs to be considered when naming the deployment:

- Deployment name length can't exceed 64 chars.

> - Two deployments with the same name created in different location will fail.

Reviewer: ```suggestion
- Two deployments with the same name created in different Azure locations (e.g. WestEurope & EastUS) will fail.
```

> - Using the same deployment name more than once, will surface only the last one in the Azure Portal.

Reviewer: ```suggestion
- Using the same deployment name more than once, will surface only the most recent deployed one in the Azure Portal.
```
